### ISSUE TITLE
[bug] Insert internal link in Firefox

### ISSUE THREAD

**Issue by WPum** [zadam/trilium#1910]
**NOTE: Seems to work fine in Chrome and desktop build**

Firefox version 88.0
app version 0.47.2 (server)

### Insert internal link in a bulleted list or numbered list
Steps:
1. type "-" or "1." then press space to create a bulleted list or numbered list
2. type "@" then select a note to create an internal link

Results:
1. the editor becomes unresponsive and any changes later wouldn't be saved
2. After refreshing the page, if you put the cursor at the next line of this internal link and press up-arrow key, the cursor will go up two lines instead of one

Frontend console log of result 1:
```
Uncaught TypeError: cyclic object value
    onerror glob.js:60
    Ie glob.js:47
    <anonymous> desktop.js:12
    <anonymous> desktop.js:2
    <anonymous> desktop.js:2
Uncaught CKEditorError: t.viewItem.parent is null
Read more: https://ckeditor.com/docs/ckeditor5/latest/framework/guides/support/error-codes.html#error-t.viewItem.parent is null
    init (index):5
    fire (index):5
    _convertItem (index):5
    convert (index):5
    toModel (index):5
    change (index):5
    toModel (index):5
    _handleContainerChildrenMutations (index):5
    handle (index):5
    init (index):5
    fire (index):5
    kn (index):5
    fire (index):5
    _onMutations (index):5
    es (index):5
    addObserver (index):5
    Gs (index):5
    qa (index):5
    kc (index):5
    Fu (index):5
    qy (index):5
    create (index):5
    create (index):5
    initEditor editable_text.js:108
    doRender editable_text.js:83
    render basic_widget.js:59
    refresh note_detail.js:117
    noteSwitched tab_aware_widget.js:61
    tabNoteSwitchedEvent tab_aware_widget.js:56
    callMethod component.js:86
    handleEvent component.js:46
    promise callback*handleEvent component.js:46
    handleEventInChildren component.js:61
    handleEvent component.js:47
    handleEventInChildren tab_caching_widget.js:73
    handleEvent component.js:47
    handleEventInChildren component.js:61
    handleEvent component.js:47
    handleEventInChildren component.js:61
    handleEvent component.js:47
    handleEventInChildren component.js:61
    handleEvent component.js:47
    handleEventInChildren component.js:61
    handleEvent component.js:47
    triggerEvent app_context.js:84
    triggerEvent component.js:53
    openTabWithNote tab_manager.js:228
    loadTabs tab_manager.js:100
    allowUpdateWithoutChange spaced_update.js:59
    loadTabs tab_manager.js:98
    start app_context.js:35
    async* desktop.js:28
    promise callback* desktop.js:26
    <anonymous> desktop.js:2
    <anonymous> desktop.js:2
```

### Insert internal link in a to-do list
Steps:
1. type "[]" then press space to create a to-do list
2. type "@" then select a note to create an internal link

Results:
1. you cannot insert a space at this time but other things seem to be fine, no error in console

**Comment by zadam** [comment]
Hi, I'm testing exact same scenario (typing `1.`, then `@` and link in FF88 on 0.47.2) and can't reproduce this. Links are inserted just fine and editor continues working ...

Is this reproducible in a blank note or only in a note with some (specific) content?

**Comment by sigaloid** [comment]
Not experiencing this issue with a to do list (created by the paragraph menu on the side) on FF 88.0 and 0.46.7 or 0.47.2 (docker).

**Comment by WPum** [comment]
> 
> 
> Hi, I'm testing exact same scenario (typing `1.`, then `@` and link in FF88 on 0.47.2) and can't reproduce this. Links are inserted just fine and editor continues working ...
> 
> Is this reproducible in a blank note or only in a note with some (specific) content?

Here are more detailed steps (I've tested):
1. Run trilium-linux-x64-server-0.47.2.tar.xz
2. Set up the new instance in FF
3. Create two notes under root A and B
4. In B type "-" + space + "@" + (select A) + **space** + anything + enter
5. editor becomes unresponsive

**Comment by chee** [comment]
@WPum is it possible you have a firefox addon enabled that could be interfering?

**Comment by zadam** [comment]
@WPum thanks, I was able to reproduce this. The critical part was adding a space after the link. I've looked into it shortly but was so far not able to figure this out.

**Comment by WPum** [comment]
Yes that's very strange.

**Comment by WPum** [comment]
> @WPum is it possible you have a firefox addon enabled that could be interfering?

Unfortunately that may not be the problem as I tested this with all addons disabled.

**Comment by WPum** [comment]
> Not experiencing this issue with a to do list (created by the paragraph menu on the side, or the `[]` shortcut) on FF 88.0 and 0.46.7 or 0.47.2 (docker).

I retested this scenerio and it's still there. Could you please try these:
1. Create a new blank note
2. type "[]" + space + "@" + (link a note) + space

**Comment by sigaloid** [comment]
You are correct, I cannot place a space there. However, it does not become unresponsive and I can still edit. It saves the changes as well with no problem. 

Copying a space manually and pasting it after moving the cursor to the end of the link does work however. :shrug:

**Comment by nemec** [comment]
I can replicate this behavior with external links as well. If you add _any_ text while the cursor sits at the end of the linkified text (and before the link arrow icon), the editor gets into this error state and cannot be recovered except by refreshing. I lose all text entered into the editor since it entered the error state. I'm also unable to use backspace to erase text while in the error state.

Example of where the cursor should be:
![Screenshot from 2021-05-30 02-08-17](https://user-images.githubusercontent.com/136378/120095643-9801c600-c0ec-11eb-9835-e1ca2890707a.png)

My install is slightly older (0.45.7), but it's the same error message(s) as in the OP. Firefox 88.0.1, Ubuntu 20.04, Server version. I cannot replicate the issue in Chromium Browser.

**Comment by lamka02sk** [comment]
I am experiencing this issue since around may when I installed version 0.47.2 also in Firefox. I used 0.44.* before. When I type \`something\` and then after the code is highlighted and I try to get out of the code block for example with right arrow key, this exact error happens in the console.
The most annoying part is that I can still write (except newline and backspace), but the changes won't be saved on the server.
The issue still persists in version 0.47.7.
